invenio_records_resources/services/records/queryparser/transformer.py

```diff
@@ -30,11 +30,6 @@ def __init__(self, mapping, allow_list, *args, **kwargs):
         self._allow_list = allow_list
         super().__init__(self, *args, **kwargs)
 
-    @classmethod
-    def factory(cls, mapping=None, allow_list=None):
```

Comment:
not needed anymore since now mapping and allow list are parameters of the query parser and passed down directly.

**IMPORTANT** this is a breaking change.